spec/unit/use_cases/save_items_details_spec.rb

```diff
@@ -0,0 +1,45 @@
+describe SaveItemsDetails do
+  let(:items_gateway) { spy }
+  let(:use_case) do
+    described_class.new(items_gateway: items_gateway)
+  end
+
+  it 'uses the items gateway to save the order details' do
+    use_case.execute(items_details: [{ part_id: '456', part_name: 'Bots', part_price: 10, part_quantity: 2 }])
+    expect(items_gateway).to have_received(:save).with([{ part_id: '456', part_name: 'Bots', part_price: 10, part_quantity: 2 }])
```

Comment:
Could assign the parts array to a variable and then refer to it twice (in execute and expect block)